src/descriptor/policy.rs

```diff
@@ -89,11 +102,29 @@ impl PkOrF {
                 key: SinglePubKey::XOnly(pk),
                 ..
             }) => PkOrF::XOnlyPubkey(*pk),
-            DescriptorPublicKey::XPub(xpub) => PkOrF::Fingerprint(xpub.root_fingerprint(secp)),
+            DescriptorPublicKey::XPub(xpub) => PkOrF::XpubOrigin(XpubOrigin {
+                fingerprint: Some(xpub.root_fingerprint(secp)),
+                derivation_path: xpub.origin.as_ref().map(|origin| origin.clone().1),
+                xkey: Some(xpub.xkey),
+            }),
         }
     }
 }
 
+/// script type used to encode multisig script
+#[derive(Debug, Clone, Copy, Serialize, Deserialize, Eq, PartialEq)]
+pub enum MultisigScriptType {
+    /// P2SH
+    #[serde(rename = "P2SH")]
+    P2sh,
+    /// P2SH-P2WSH
+    #[serde(rename = "P2SH_P2WSH")]
+    P2shP2wsh,
+    /// P2WSH
+    #[serde(rename = "P2WSH")]
+    P2wsh,
+}
```

Comment:
We can use `ScriptType` for any type of scripts.. Not just multisigs..

Reply:
Yep. Good point. I assume you're recommending the Enum be renamed to `ScriptType`? And maybe `P2tr should be included here now too if we're moving this to policy?

Reply:
Yup sounds good.. I think bdk can already recognize `p2tr`. A test with an example taproot descriptor into policy and checking if its parsing it correctly would be really nice..